src/net/sourceforge/kolmafia/session/ChoiceAdventures.java

```diff
@@ -8969,10 +8969,209 @@ else if (KoLCharacter.isEd()) {
           result[2] = new ChoiceOption("Walk away in disappointment");
           return result;
         }
+
+      case 1499:
+        {
+          // A Labyrinth of Shadows
+
+          String responseText = ChoiceManager.lastResponseText;
+          Map<Integer, String> choices = ChoiceUtilities.parseChoices(responseText);
+          int options = choices.size();
+
+          // 1 = "Press on"
+          // 2 = (random theme)
+          // 3 = (random theme)
+          // 4 = (random theme)
+          // 5 = "Turn in a circle"
+          // 6? = "Leave this place"
+
+          result = new ChoiceOption[options];
+          result[0] = new ChoiceOption("Randomize themes");
+          for (int i = 1; i <= 3; ++i) {
+            result[i] = shadowLabyrinthSpoiler(choices.get(i));
+          }
+          result[4] = new ChoiceOption("Randomize themes");
+          result[5] = new ChoiceOption("Waste a turn");
```

Comment:
This NC might not actually take a turn, even if you leave, so maybe "Leave with nothing" if that's the case?

Reply:
Done.